qiskit/visualization/transition_visualization.py

```diff
@@ -72,10 +72,10 @@ def __mul__(self, b):
             return self._multiply_with_quaternion(b)
         elif isinstance(b, (list, tuple, np.ndarray)):
             if len(b) != 3:
-                raise Exception(f"Input vector has invalid length {len(b)}")
+                raise RuntimeError(f"Input vector has invalid length {len(b)}")
             return self._multiply_with_vector(b)
         else:
-            raise Exception(f"Multiplication with unknown type {type(b)}")
+            raise TypeError(f"Multiplication with unknown type {type(b)}")
```

Comment:
This should likely `return NotImplemented` - Python will manually raise a suitable `TypeError` if the mirrored multiplication (`type(b).__rmul__(b, self)`) fails as well, but raising an exception means the mirror multiplication never gets attempted.